ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -300,6 +308,10 @@ private module Cached {
     TSynthHashSplatParameterNode(DataFlowCallable c) {
       isParameterNode(_, c, any(ParameterPosition p | p.isKeyword(_)))
     } or
+    TSynthSplatParameterNode(DataFlowCallable c) {
+      exists(c.asCallable()) and // exclude library callables
```

Comment:
why?

Reply:
> I initially thought it wouldn't be useful to have splat flow into library callables since they have no real method bodies, but of course they can still propagate flow from their positional params and so they shouldn't necessarily be excluded here.

We want to make sure that library callables involving positional summaries also work when arguments are passed in through a splat argument. E.g. if `returnFst` is modeled to return its first positional parameter, then we want there to be flow from `x` to the result of `returnFst(*[x,y])`.

> But the other issue is I got some consistency test failures when I didn't exclude them

Ok, it looks like we actually need to synthesize a post-update splat parameter node as well (the same goes for hash splat parameters). Consider:

```rb
def foo(x, y)
  x[0] = y
end

a = [0]
args = [a, taint]
foo(*args);
sink(args[0][0])
```

I suggest we do this follow-up, so let's keep the restriction for library callables for now.